projectZoo/src/pr1/Zoo.java

```diff
@@ -1,4 +1,181 @@
+    } 
+ 
+    void answerEqw5() {
+        buyTicket();
+    }
+
+    void mindChanging() {
+        Scanner sc = new Scanner(System.in);
+        String buyAnswer = sc.nextLine();
+        buyAnswer = buyAnswer.toLowerCase();
+        if(buyAnswer.equals("y")){
+            m.menu();
+        }else if(buyAnswer.equals("n")){
+            actualBuyTicket();
+        }else{
+            System.out.println("Please enter [Y/N]");
+            mindChanging();
+
+        }
+    }
+
+    void getVisitorInfo() {
+        System.out.println("Please enter your name: "); //ім'я
+        p.name = sc.nextLine();
+
+        System.out.println("And surname: ");            //прізвище
+        p.surname = sc.nextLine();
+
+        System.out.println("Good! Now please enter the serial number of your passport");   //паспорт
+        p.passportNumber = sc.nextLine();
+
+        System.out.println("Ooh, pretty! Now please enter your year of birth! [YYYY]"); //дата
+        p.yearOB= sc.nextInt();
+        System.out.println("Month of birth! [MM]");
+        p.monthOB = sc.nextInt();
+        System.out.println("And the day of birth! [DD]");
+        p.dayOB = sc.nextInt();
+
+        LocalDate fullBD = LocalDate.of(p.yearOB, Month.of(p.monthOB), p.dayOB);
+//_______________________________________________________
+        if (calculateAge(fullBD, currentDate)<18){
+            System.out.println("Sorry, you're not allowed no take a part in out show(((");
+        }
+        else {
+           actualBuyTicket();
+        }
+//_______________________________________________________
+
+    }
+
+    void actualBuyTicket(){
+        System.out.println("Enter the number of tickets you want to buy!");
+        int numOfBuyTickets = sc.nextInt();
+
+        if (numOfBuyTickets > quantity) {
+            System.out.println("We don't have so many tickets already! Would you like to buy some less?");
+            actualBuyTicket();
+        } else if (numOfBuyTickets == 0) {
+            System.out.println("Uhh...Have you changed you're mind?[y/n]");
+            mindChanging();
+        } else if (numOfBuyTickets < 0) {
+            System.out.println("That's not serious...");
+            actualBuyTicket();
+        } else {
+            System.out.println("There was " + quantity + " tickets");
+            quantity = quantity - numOfBuyTickets;
+            System.out.println("And now there is " + quantity + " of them");
+            System.out.println("You've bought " + numOfBuyTickets + " ticket(s). Thank you!");
+
+            System.out.println("*** a couple of people arrived and also bought some tickets ***");
+            System.out.println("Okay, we're ready to start our show. Take your places and enjoy!");
+            try {
+                first.announce();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    void setVisitorsList(){
+        p.countOfVisitors = 10-quantity;
```

Comment:
Magic number too

Reply:
```suggestion
        p.countOfVisitors = Const.quantityOfAllVisitors-quantityOfTickets;
```
You're right, now it's much easier to understand!